src/centralserver/admin-service/src/main/java/org/niis/xroad/centralserver/restapi/repository/GlobalGroupMemberRepository.java

```diff
@@ -0,0 +1,117 @@
+
+    @Builder
+    @Getter
+    class Criteria {
+        private final Integer groupId;
+        private final String query;
+        private final String memberClass;
+        private final String instance;
+        private final List<String> codes;
+        private final List<String> subsystems;
+        private final List<XRoadObjectType> types;
+    }
+
+    static Specification<GlobalGroupMember> findSpecification(Criteria criteria) {
+        return (root, query, builder) -> findPredicate(root, builder, criteria);
+    }
+
+    private static Predicate findPredicate(Root<GlobalGroupMember> root, CriteriaBuilder builder, Criteria criteria) {
+        final Join<GlobalGroupMember, ClientId> member = root.join(GlobalGroupMember_.identifier);
+        final Join<GlobalGroupMember, GlobalGroup> globalGroup = root.join(GlobalGroupMember_.globalGroup);
+
+        var pred = builder.equal(globalGroup.get(GlobalGroup_.id), criteria.getGroupId());
+        if (StringUtils.isNotBlank(criteria.getQuery())) {
+            pred = builder.and(pred, searchPredicate(root, member, builder, criteria));
+        }
+        if (StringUtils.isNotBlank(criteria.getMemberClass())) {
+            pred = builder.and(pred, builder.equal(member.get(XRoadId_.memberClass), criteria.getMemberClass()));
+        }
+        if (StringUtils.isNotBlank(criteria.getInstance())) {
+            pred = builder.and(pred, builder.equal(member.get(XRoadId_.xRoadInstance), criteria.getInstance()));
+        }
+        if (!CollectionUtils.isEmpty(criteria.getCodes())) {
+            pred = builder.and(pred, member.get(XRoadId_.memberCode).in(criteria.getCodes()));
+        }
+        if (!CollectionUtils.isEmpty(criteria.getSubsystems())) {
+            pred = builder.and(pred, member.get(XRoadId_.subsystemCode).in(criteria.getSubsystems()));
+        }
+        if (!CollectionUtils.isEmpty(criteria.getTypes())) {
+            pred = builder.and(pred, member.get(XRoadId_.objectType).in(criteria.getTypes()));
+        }
+        return pred;
+    }
+
+    private static Predicate searchPredicate(Root<GlobalGroupMember> root, Join<GlobalGroupMember, ClientId> member,
+                                             CriteriaBuilder builder,
+                                             Criteria criteria) {
+        return builder.or(
+                caseInsensitiveLike(root, builder, criteria.getQuery(), member.get(XRoadId_.MEMBER_CODE)),
+                caseInsensitiveLike(root, builder, criteria.getQuery(), member.get(XRoadId_.MEMBER_CLASS)),
+                caseInsensitiveLike(root, builder, criteria.getQuery(), member.get(XRoadId_.SUBSYSTEM_CODE)),
+                caseInsensitiveLike(root, builder, criteria.getQuery(), member.get(XRoadId_.X_ROAD_INSTANCE)),
+                caseInsensitiveLike(root, builder, criteria.getQuery(), member.get(XRoadId_.OBJECT_TYPE))
+        );
+    }
+
+    @Query("SELECT m FROM #{#entityName} m JOIN m.globalGroup g "
```

Comment:
if you would rename this method to `findByGlobalGroupId`,  then it doesn't need `@Query` annotation.

Also this query could be simplified like this:
```sql 
"SELECT m FROM GlobalGroupMember m WHERE m.globalGroup.id = :groupId"
```
Use `#{#entityName}` only if you need dynamic entity name (e.g. in abstract base class). Otherwise you loose IDE support for autocomplete and syntax checking/highlighting.